utils/hct/hctdb.py

```diff
@@ -7656,17 +7656,34 @@ def build_valrules(self):
         self.add_valrule(
             "Sm.TGSMUnsupported", "Thread Group Shared Memory not supported %0."
         )
-        self.add_valrule(
+        self.add_valrule_msg(
             "Sm.WaveSizeValue",
-            "Declared WaveSize %0 outside valid range [%1..%2], or not a power of 2.",
+            "WaveSize value must be a power of 2 in range [4..128]",
+            "WaveSize %0 (%1) outside valid range [%2..%3], or not a power of 2.",
```

Comment:
Is there a good reason that we merge these two diagnostic messages into one? I'm just curious why we don't have a diagnostic specifically for when a value is not a power of 2.

Reply:
They weren't separate before, so I'm didn't separate them in this PR.

For validation, I don't see much value separating them.  If you want separate diagnostics for the normal path, that would require separating them in DiagnosticSemaKinds.td.  Still, I don't see much value in doing that.